pkg/types/vsphere/validation/platform.go

```diff
@@ -94,3 +99,41 @@ func validateFolder(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
 
 	return allErrs
 }
+
+// ValidateVCenterAddress validates vCenter to only contain hostname and port.
+func ValidateVCenterAddress(vCenter string) error {
+	if strings.TrimSpace(vCenter) == "" {
+		return errors.New("must specify the name of the vCenter")
+	}
+	vCenterURL, err := url.ParseRequestURI(vCenter)
+	if err == nil && vCenterURL.Host != "" {
+		return errors.Errorf("vCenter hostname cannot contain url scheme")
+	}
+
+	vCenterURL, err = url.Parse(vCenter)
+	if err != nil {
+		return errors.Errorf("vCenter hostname is not valid")
+	}
```

Comment:
I'm not sure I understand why you are doing this (checking for hostname validity twice--you check again below). I know I suggested something like this, so I'm not sure if my suggestion was unclear. This is not what I had in mind. I was suggesting that INSTEAD of the manual checking host and port, you could leverage `url.Parse` or `url.ParseRequestURI`. 

The entire flow would look something like this:
1. Check if vCenter contains a scheme if it does return an error (you are already doing this step)
2. We know vCenter does not contain a scheme, so create a temporary variable and prepend the scheme to vcenter
3. Run that temporary variable through one of the parse methods, which will then check for correct domain and port
4. if there is an error make sure to replace the temporary variable with the actual value passed

I have created a playground to illustrate how parse checks the domain and port: https://play.golang.org/p/S_CutN804xt

You are free to use this idea--and it could potentially be improved so that the checks in step 3 are run even if the step 1 check fails, but we could discuss that later.

The other checks you have already included are also fine, but I'm not sure why this `Parse` is here.

Reply:
Yeah your approach is more understandable. I will modify it.